src/msg/producer/types.go

```diff
@@ -154,9 +157,15 @@ func NewFilterFuncMetadata(
 	return FilterFuncMetadata{
 		FilterType: filterType,
 		SourceType: sourceType,
+		cacheKey:   fmt.Sprintf("%s::%s", filterType.String(), sourceType.String()),
```

Comment:
nit: i think string concat usually is a little more performant and I think this is simple enough that we don't need the overhead of Sprintf. Can we switch to using `filterType.String() + "::" + sourceType.String()`